common/src/app/common/geom/shapes/transforms.cljc

```diff
@@ -485,205 +493,234 @@
                                 %)))
     shape))
 
+(defn -transform-shape
+  [shape {:keys [round-coords?]
+          :or {round-coords? true}}]
+  (if (and (contains? shape :modifiers) (empty-modifiers? (:modifiers shape)))
+    (dissoc shape :modifiers)
+    (let [shape     (apply-displacement shape)
+          center    (gco/center-shape shape)
+          modifiers (:modifiers shape)]
+      (if (and (not (empty-modifiers? modifiers)) center)
+        (let [transform (modifiers->transform center modifiers)]
+          (-> shape
+              (set-flip modifiers)
+              (apply-transform transform round-coords?)
+              (apply-text-resize modifiers)
+              (dissoc :modifiers)))
+        shape))))
+
+(def transform-shape* (memoize -transform-shape))
```

Comment:
I have a concern here, I think that, in depending on the usage, we can increase the memory usage without a way to liberate. It is a good first step, but I think we will need to setup a "limit" on objects we can track. We want something in the middle of memoize and "core.cache". I will think on it and prepare a solution, for now this is just a NOTE.

Reply:
I think you're right. But I haven't seen any options like `core.cache` but for the frontend.